File: src/main/java/messages_control/TextListener.java
package messages_control;

import dbLogAddon.handlers.LogRepositoryImpl;
import dbLogAddon.model.Message;
import lombok.extern.log4j.Log4j;
import net.dv8tion.jda.core.events.message.MessageReceivedEvent;
import net.dv8tion.jda.core.hooks.ListenerAdapter;

@Log4j(topic = "file")
public class TextListener extends ListenerAdapter {
    private LogRepositoryImpl logRepository = new LogRepositoryImpl();

    @Override
    public void onMessageReceived(MessageReceivedEvent event) {
        String command;
        if (event.getAuthor().isBot()) {
            return;
        }
        logRepository.saveMassage(new Message(event));
        command = event.getMessage().getContentRaw();

        if (MessageAnalyze.isCommandToBot(command)) {
            MessageAnalyze analyze = new MessageAnalyze(event);
            event.getChannel().sendMessage(analyze.makeMessage()).queue();
        }
    }
}

File: src/main/scala/listener/TextListener.scala
package listener

import analyzer.MessageAnalyzer
import commands.DiceCommand
import dbLogAddon.handlers.LogRepositoryImpl
import dbLogAddon.model.Message
import net.dv8tion.jda.api.events.message.MessageReceivedEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter

class TextListener extends ListenerAdapter {
  private val logRepository = new LogRepositoryImpl
  private val msgAnalyzer = new MessageAnalyzer()

  override def onMessageReceived(event: MessageReceivedEvent): Unit = {
    if (!event.getAuthor.isBot) {

      //Maybe we don't need it anymore
      //logRepository.saveMassage(new Message(event))

      event match {
        case ev if MessageAnalyzer.isCommandToBot(ev) =>
          val response = msgAnalyzer.analyzeEventAndMakeResponse(event)
          event.getChannel.sendMessage(response).queue()

        case ev if MessageAnalyzer.isDiceRollCommand(ev) =>
          val response = new DiceCommand().d(ev)
          event.getChannel.sendMessage(response).queue()

        case _ => ()
      }
    }
  }
}
